sql/expression/function/reverse_repeat_replace.go

```diff
@@ -0,0 +1,252 @@
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+// Reverse is a function that returns the reverse of the text provided.
+type Reverse struct {
+	expression.UnaryExpression
+}
+
+// NewLower creates a new Lower expression.
+func NewReverse(e sql.Expression) sql.Expression {
+	return &Reverse{expression.UnaryExpression{Child: e}}
+}
+
+// Eval implements the Expression interface.
+func (r *Reverse) Eval(
+	ctx *sql.Context,
+	row sql.Row,
+) (interface{}, error) {
+	v, err := r.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if v == nil {
+		return nil, nil
+	}
+
+	v, err = sql.Text.Convert(v)
+	if err != nil {
+		return nil, err
+	}
+
+	return reverseString(v.(string)), nil
+}
+
+func reverseString(s string) string {
+	r := []rune(s)
+	for i, j := 0, len(r)-1; i < len(r)/2; i, j = i+1, j-1 {
```

Comment:
I think it's better to have a `j < j` condition instead of calculating every time length of string and divide by 2.